### DISCUSSION TITLE
[Ethereum] Signing with Prefix

### DISCUSSION THREAD

**Discussion by LayneHaber** [trustwallet/wallet-core#1619] [Q&A]
I have an SDK that uses the `ethers.Signer` abstract class as an interface. I expect the messages returned from `.signMessage` to include the ethereum prefix, but it appears when using an injected Trust Wallet provider this is not the case. 

What is the recommended way for handling this and adding the prefix with the `ethers` package?

Sorry if this is not the best place/way to ask this question!

**Comment by optout21** [comment]
Can you elaborate on Ethereum project, and Trust Wallet as provider?  I can't fully understand.  Do you have shareable sample code?

**Comment by hewigovens** [comment]
Do you mean trust web3 provider?

**Comment by LayneHaber** [comment]
Someone is using an injected trust wallet as the web3 provider, yes
